### ISSUE TITLE
TreeSelect & ReactiveForm: Incorrect Behavior on Marking Controller as Touched

### ISSUE THREAD

**Issue by miladFaghihiEdiven** [primefaces/primeng#17723]
### Describe the bug

The `TreeSelect` component incorrectly marks the form control as touched when the overlay is opened.

**Why This Is a Problem?**
Typically, error messages should be displayed when a form control is invalid and touched. However, with the current `TreeSelect` behavior, opening the overlay immediately triggers validation errors, which is not the expected UX.

It's because as soon as we open the TreeSelect overlay, the FormControl is marked as touched, and not when the overlay is closed or an item is selected.

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/mwanuzbv?file=src%2Fapp%2Ftree-select-reactive-forms-demo.html

### Environment

PrimeNg 19.0.5

### Angular version

19.0.0

### PrimeNG version

v19

### Steps to reproduce the behavior

- Open the treeSelect overlay.
- The FormControl will mark immediately as touched.

### Expected behavior

The `FormControl` must be marked as touched when:

- The overlay is closed.
- A user selects an item.

**Comment by miladFaghihiEdiven** [comment]
any update? it was planed for v19.1.1, right now v20 has released but no fix for the bug

**Comment by thoekstraorama** [comment]
We're experiencing the same problem. Is there any update for this bug?

**Comment by moraisacr** [comment]
Hi everyone 👋  
I’ve submitted a fix for this issue and linked it above.

The problem was that `onModelTouched()` was being called when the overlay opened.  
Now it is triggered only when the overlay is closed, which aligns with Angular form expectations.

PR: #19264  
Thanks for the reports and patience.
